torchvision/models/alexnet.py

```diff
@@ -1,5 +1,5 @@
 import torch.nn as nn
-import torch.utils.model_zoo as model_zoo
+from torch import hub
```

Comment:
This will break with PyTorch 1.0.1 and older.
Should we care or would we need to add backcompat fixes?

ccing @soumith for advice

Reply:
We don't _need_ to use `torch.hub`, its just what [you originally suggested](https://github.com/pytorch/vision/issues/862#issuecomment-485955748). If we don't want to break backward compatibility we have two options:
1. We could roll back to `torch.utils.model_zoo.load_url`, which simply imports `torch.hub.load_state_dict_from_url` for `torch > 1.0.1`,

or

2. we could put something along these lines on top of each module:

```python
import torch
if torch.__version__ > '1.0.1':
    load_state_dict_from_url = torch.utils.model_zoo.load_url
else:
    load_state_dict_from_url = torch.hub.load_state_dict_from_url
```

I would opt for option 1. for now and only deal with this when `torch.utils.model_zoo` is deprecated / removed in a stable release of `torch`.

Reply:
Here is another option:
```python
# in model/utils.py
try:
    from torch import hub as ...
    _HAS_HUB = True
except ImportError:
    from torch.utils.model_zoo import ...
    _HAS_HUB = False

def load_state_dict(...):
    # make it accept a dummy progress if _HAS_HUB is False
```
and then we just import from `.models.utils`. What do you think?

Reply:
`torch.utils.model_zoo.load_url` has a `progress` flag since version `0.4.0`. Are we looking for BC for earlier versions? If not, I see no point in the `_HAS_HUB` flag.

Reply:
No need for the `_HAS_HUB` then, I didn't like it either :-)

Reply:
So we are going with option 1 ?

Reply:
I'd rather have my modified option 2, because once `torch.utils.model_zoo` gets removed, we will have to do more work to update torchvision with option 1 :-)

Reply:
Done